app/monitoringapi.go

```diff
@@ -76,32 +87,46 @@ func wireMonitoringAPI(life *lifecycle.Manager, addr string, localNode *enode.Lo
 	return nil
 }
 
-// newReadyHandler returns a http.HandlerFunc which returns 200 when both the beacon node is synced and all quorum peers can be pinged  in parallel within a timeout. Returns 500 otherwise.
-func newReadyHandler(tcpNode host.Host, eth2Cl eth2client.NodeSyncingProvider, peerIDs []peer.ID) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		ctx, cancel := context.WithTimeout(r.Context(), time.Second)
-		defer cancel()
-
-		var ready float64
-		defer func() { readyzGauge.Set(ready) }()
-
-		syncing, err := beaconNodeSyncing(ctx, eth2Cl)
-		if err != nil {
-			writeResponse(w, http.StatusInternalServerError, "Failed to get beacon sync state")
-			return
-		} else if syncing {
-			writeResponse(w, http.StatusInternalServerError, "Beacon node not synced")
-			return
+// startReadyChecker returns function which returns an error resulting from ready checks periodically.
+func startReadyChecker(ctx context.Context, tcpNode host.Host, eth2Cl eth2client.NodeSyncingProvider, peerIDs []peer.ID, clock clockwork.Clock) func() error {
+	var (
+		mu       sync.Mutex
+		readyErr error
+	)
+	go func() {
+		ticker := clock.NewTicker(10 * time.Second)
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.Chan():
+				mu.Lock()
```

Comment:
remember to decrease mutex locked scope to absolute minimum, and remember to NEVER do IO (disk/network) calls while a lock is held.